app/controllers/song_categories_controller.rb

```diff
@@ -0,0 +1,22 @@
+class SongCategoriesController < ApplicationController
+  load_permissions_and_authorize_resource
+
+  def index
+    @song_categories = initialize_grid(SongCategory.all, order: 'name', order_direction: 'asc')
+  end
+
+  def show #???
```

Comment:
You don't need this if you don't want but it would be used to show all songs under a specific category. But it might be easier to just filter on the category in the index view.

Reply:
On second thought, you don't have any views for `song_categories` for non-admins so the index method won't be used either. You might want to consider removing the whole controller since the end-user will probably only use the categories in the song view for filtration. What do you think?

Reply:
You're absolutely right! The controller is unnecessary and I will remove it.